incubator/monochart/templates/statefulset.yaml

```diff
@@ -61,8 +65,8 @@ spec:
           protocol: {{ default "TCP" $port.protocol  }}
 {{- end }}
         volumeMounts:
-        - mountPath: {{ .Values.persistence.mountPath | quote }}
-          name: storage
+        - mountPath: {{ .Values.statefulset.persistence.mountPath | quote }}
```

Comment:
add `      {{- if .Values.statefulset. persistence.enabled }}`

Reply:
ok, so if we add the flag `statefulset.persistence.enabled` and set it to `false`, then `volumeClaimTemplates` would not be created and `StatefulSet` would not have a `volumeMount`. But how useful the `StatefulSet` would be in that case? The purpose of `StatefulSet` is to have some persistent storage.

So, if we add `statefulset.persistence.enabled` and set it to `false`, should we add the logic to use `persistentVolumeClaim`?

```
{{- if not .Values.statefulset.persistence.enabled }}
      - name: {{ (include "common.fullname" . ) }}
        persistentVolumeClaim:
          claimName: {{ .Values.persistence.existingClaim | default (include "common.fullname" . ) }}
  {{- else  }}
  volumeClaimTemplates:
```

@goruha  ?

Reply:
add the same if\else for mount also

Reply:
added `statefulset.persistence.useVolumeClaimTemplates` instead of `statefulset.persistence.enabled` since persistence should always be enabled for `StatefulSet` - that's the purpose of having `StatefulSet` (instead of e.g. `Deployment`).

If `statefulset.persistence.useVolumeClaimTemplates` is set to `true`, we use `volumeClaimTemplates`.

If `statefulset.persistence.useVolumeClaimTemplates` is set to `false`, we use `persistentVolumeClaim` in `volumes`.

-----------------------------------------------------------------------------
We always have `mountPath` for the volume (without any if/else conditions) regardless of whether `statefulset.persistence.useVolumeClaimTemplates` is set to `true` or `false`:

```
        volumeMounts:
        - mountPath: {{ .Values.statefulset.persistence.mountPath | quote }}
```

because that's the purpose of `StatefulSet` - to always have a persistent volume

@goruha ^

@osterman, what do you think about ^ ?

Reply:
Yes, there's almost zero reason to implement boolean for `StatefulSet` because the overwhelming reason to use a Statefulet is for Persistance. Let's not add it if we don't need it now. We can add it later.

Reply:
We need to get this PR merged. The more feature flags we add, the greater the risk.